src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -322,6 +332,76 @@ private async Task IndexAddressesContinuouslyAsync()
             }
         }
 
+        private void RewindOutPointRepository(uint256 hashToReorg)
+        {
+            Block blockToReorg = this.consensusManager.GetBlockData(hashToReorg).Block;
+
+            // For efficiency, only do one lookup to the block store. So we accumulate all needed transactions first.
+            var previousTransactionsToRetrieve = new HashSet<uint256>();
+
+            foreach (Transaction tx in blockToReorg.Transactions)
+            {
+                if (tx.IsCoinBase)
+                    continue;
+
+                foreach (TxIn input in tx.Inputs)
+                {
+                    previousTransactionsToRetrieve.Add(input.PrevOut.Hash);
```

Comment:
Are you getting the `PrevOut` here to get the input that was spent?

Reply:
Yes, specifically, we need its amount and its scriptPubKey. The only way to do that is to interrogate the transaction index. That is because we have already deleted the outpoint from the repository upon it initially being spent. So we have to reconstitute the data when rewinding.

Reply:
OK cool, was really just asking to check if my own understanding of this was correct ;)